### PR TITLE
API - service : GET Usvc veraion

### PR DESCRIPTION
**Pull request by mithunbharadwaj**

### CODE REVIEW

File: asab/api/service.py [hunk 1]
```diff
@@ -23,6 +25,21 @@ def __init__(self, app, service_name="asab.ApiService"):
 		self.ZkContainer = None
 		self.AttentionRequired = {}  # dict of errors found.
 
+		path = asab.Config.get("general", "manifest_json_path")
```
Comment (ateska) [inline asab/api/service.py:28]: `asab.Config.get("general", "manifest")`

File: asab/api/service.py [hunk 2]
```diff
@@ -23,6 +25,21 @@ def __init__(self, app, service_name="asab.ApiService"):
...
+		if not os.path.isfile(path):
+			if os.path.isfile("/MANIFEST.json"):
+				path = "/MANIFEST.json"
+			elif os.path.isfile("MANIFEST.json"):
+				path = "MANIFEST.json"
+			else:
+				L.warning("MANIFEST.json file not found")
+		try:
```
Comment (ateska) [inline asab/api/service.py:36]: Manifest could be not provided (empty string) which is fine and should not generate warning

File: asab/api/service.py [hunk 3]
```diff
@@ -23,6 +25,21 @@ def __init__(self, app, service_name="asab.ApiService"):
 		self.ZkContainer = None
 		self.AttentionRequired = {}  # dict of errors found.
 
+		path = asab.Config.get("general", "manifest_json_path")
+		if not os.path.isfile(path):
```
Comment (ateska) [inline asab/api/service.py:29]: `if len(path) == 0`

**Review by ateska — CHANGES_REQUESTED** [review]

File: asab/api/service.py [hunk 4]
```diff
@@ -23,6 +25,19 @@ def __init__(self, app, service_name="asab.ApiService"):
...
 		self.AttentionRequired = {}  # dict of errors found.
 
+		path = asab.Config.get("general", "manifest")
+		if len(path) == 0:
+			if os.path.isfile("/MANIFEST.json"):
+				path = "/MANIFEST.json"
+			elif os.path.isfile("MANIFEST.json"):
+				path = "MANIFEST.json"
+		try:
+			with open(path) as f:
```
Comment (ateska) [inline asab/api/service.py:35]: `if len(path) == 0: manifest = dict`

This code HAS TO run on systems, where there is no MANIFEST without error.

**Review by ateska — CHANGES_REQUESTED** [review]

File: asab/api/service.py (hunk 4 above)
Comment (mithunbharadwaj) [reply asab/api/service.py:35]: @ateska I throw this warning if its missng

Loading MANIFEST.json failed for reason [Errno 2] No such file or directory: '/MANIFEST.json'

Do you think it is the correct action?

File: asab/api/service.py [hunk 5]
```diff
@@ -13,15 +15,33 @@
 
 ##
 
+asab.Config.add_defaults({
+	"general": {
+		"manifest": "/MANIFEST.json",
```
Comment (mithunbharadwaj) [inline asab/api/service.py:20]: @ateska maybe this should just be empty string ?

Comment (ateska) [reply asab/api/service.py:20]: yes

File: asab/api/service.py (hunk 4 above)
Comment (ateska) [reply asab/api/service.py:40]: Let's have a quick call on this.
We  need to maintain backward compatibility without any warnings.

Comment (mithunbharadwaj) [reply asab/api/service.py]: @ateska Sure when can we have the call?

File: asab/api/service.py [hunk 6]
```diff
@@ -41,7 +41,7 @@ def __init__(self, app, service_name="asab.ApiService"):
 				with open(path) as f:
 					self.Manifest = json.load(f)
 			except Exception as e:
-				L.exception(e)
+				L.log(e)
```
Comment (ateska) [inline asab/api/service.py:44]: This is not correct.
`L.exception("Error when reading manifest")` ... is correct.

**Review by ateska — APPROVED** [review]

**Pull request merged**